subscriptions-api.md

```diff
@@ -0,0 +1,613 @@
+("push"-style) require a mechanism to configure such subscriptions on the
+subscription manager.
+
+The subscription manager API defines a subscription object and an API for
+creating, updating and deleting subscriptions on a subscription manager.
+
+The protocol used to configure the subscription manager is decoupled from the
+delivery protocol, meaning that an application can configure a push delivery
+into an AMQP destination using an API call initiated over HTTP.
+
+A subscription manager that natively exists in a middleware implementation might
+also have a separate CloudEvents subscription manager API endpoint
+implementation. If such a separate API endpoint exists and creating
+subscriptions is not feasible on such a separate API endpoint, the respective
+operation might not be available. For instance, creating an MQTT subscription is
+not feasible from outside an MQTT connection, but an existing subscription may
+be enumerable through a subscription collection and deleting the subscription
+from the collection might terminate it.
+
+#### 3.2.1. Subscription Object
+
+A subscription manager manages a collection of subscriptions. The upper limit on
+how many subscriptions are supported is implementation specific.
+
+Each subscription is represented by an object that has the following properties:
+
+- **id** (string) – REQUIRED. The unique identifier of the subscription in the
+  scope of the subscription manager.
+- **protocol** (string) - REQUIRED. Identifier of a delivery protocol. Because of
+  WebSocket tunneling options for AMQP, MQTT and other protocols, the URI
+  scheme is not sufficient to identify the protocol. The protocols with
+  existing CloudEvents bindings are identified as "AMQP", "MQTT3", "MQTT5",
+  "HTTP", "KAFKA", and "NATS". An implementation MAY add support for further
+  protocols.
+- **protocolsettings** (map) - OPTIONAL. A set of settings specific to the
+  selected delivery protocol provider. Options for those settings are listed
+  in the following subsection. An implementation can offer more options.
+  Examples for such settings are credentials, which generally vary by
+  transport, rate limits and retry policies, or the QoS mode for MQTT.
+  See the [Protocol Settings](#322-protocol-settings) section for further details.
+- **sink** (URI) - REQUIRED. The address to which events shall be delivered using
+  the selected protocol. The format of the address MUST be valid for the
+  selected protocol or one of the protocol's own transport bindings (e.g. AMQP
+  over WebSockets).
+- **filter** - OPTIONAL. A filter is an expression of a particular filter
+  dialect that evaluates to true or false and that determines whether an
+  instance of a cloud event should be delivered to the subscription's sink. If
+  a filter expression evaluates to false, the event MUST NOT be sent to the
+  sink. Support for particular filter dialects might vary across different
```

Comment:
do we want to mandate what happens in the case of it evaluating to 'true'? While we can't say it MUST be delivered, we can probably say that it "MUST be attempted to be delivered", or something like that.

Reply:
+1